### ISSUE TITLE
Summary of papers on distributed training in nips 2018

### ISSUE THREAD

**Issue by yuyicg** [sql-machine-learning/elasticdl#196]
**flip through and summarize the papers on distributed training in nips 2018, convenient for us to see if there are any insights we can learn from.**

[PCA of high dimensional random walks with comparison to neural network training](http://papers.nips.cc/paper/8232-pca-of-high-dimensional-random-walks-with-comparison-to-neural-network-training.pdf)
one technique to visualize the training of neural networks is to perform PCA on the parameters over the course of training and to project to the subspace spanned by the first few PCA components. In this paper we compare this technique to the PCA of a high dimensional random walk.

[Evolutionary Stochastic Gradient Descent for Optimization of Deep Neural Networks](http://papers.nips.cc/paper/7844-evolutionary-stochastic-gradient-descent-for-optimization-of-deep-neural-networks.pdf)
this paper propose a population-based Evolutionary Stochastic Gradient Descent (ESGD) framework for optimizing deep neural networks. ESGD combines SGD and gradient-free evolutionary algorithms as complementary algorithms in one framework. 

[Step Size Matters in Deep Learning](http://papers.nips.cc/paper/7603-step-size-matters-in-deep-learning.pdf)
to elucidate the effects of the step size on training of neural networks, we study the gradient descent algorithm as a discrete-time dynamical system, and by analyzing the Lyapunov stability of different solutions, we show the relationship between the step size of the algorithm and the solutions that can be obtained with this algorithm.

[How Does Batch Normalization Help Optimization?](http://papers.nips.cc/paper/7515-how-does-batch-normalization-help-optimization.pdf)
In this work, we demonstrate that such distributional stability of layer inputs has little to do with the success of BatchNorm. Instead, we uncover a more fundamental impact of BatchNorm on the training process: it makes the optimization landscape significantly smoother. This smoothness induces a more predictive and stable behavior of the gradients, allowing for faster training.

[Loss Surfaces, Mode Connectivity, and Fast Ensembling of DNNs](http://papers.nips.cc/paper/8095-loss-surfaces-mode-connectivity-and-fast-ensembling-of-dnns.pdf)
we propose a new ensembling method entitled Fast Geometric Ensembling (FGE). Using FGE we can train high-performing ensembles in the time required to train a single model. We achieve improved performance compared to the recent state-of-the-art Snapshot Ensembles, on CIFAR-10, CIFAR-100, and ImageNet.

[Bayesian Distributed Stochastic Gradient Descent](http://papers.nips.cc/paper/7874-bayesian-distributed-stochastic-gradient-descent.pdf)
proposed algorithm mitigates the straggler effect in synchronous, gradient-based optimization by choosing an optimal cutoff beyond which mini-batch gradient messages from slow workers are ignored. 

[BML: A High-performance, Low-cost Gradient Synchronization Algorithm for DML Training](http://papers.nips.cc/paper/7678-bml-a-high-performance-low-cost-gradient-synchronization-algorithm-for-dml-training.pdf)
In this paper we propose BML, a new gradient synchronization algorithm with higher network performance and lower network cost than the current practice

[L4: Practical loss-based stepsize adaptation for deep learning](http://papers.nips.cc/paper/7879-l4-practical-loss-based-stepsize-adaptation-for-deep-learning.pdf)
we propose a stepsize adaptation scheme for stochastic gradient descent.

[Co-teaching: Robust training of deep neural networks with extremely noisy labels](http://papers.nips.cc/paper/8072-co-teaching-robust-training-of-deep-neural-networks-with-extremely-noisy-labels.pdf)
in this paper, we propose a new deep learning paradigm called ''Co-teaching'' for combating with noisy labels.

[Visualizing the Loss Landscape of Neural Nets](http://papers.nips.cc/paper/7875-visualizing-the-loss-landscape-of-neural-nets.pdf)
In this paper, we explore the structure of neural loss functions, and the effect of loss landscapes on generalization, using a range of visualization methods.

[Mesh-TensorFlow: Deep Learning for Supercomputers](http://papers.nips.cc/paper/8242-mesh-tensorflow-deep-learning-for-supercomputers.pdf)
we introduce Mesh-TensorFlow, a language for specifying a general class of distributed tensor computations.

[GradiVeQ: Vector Quantization for Bandwidth-Efficient Gradient Aggregation in Distributed CNN Training](http://papers.nips.cc/paper/7759-gradiveq-vector-quantization-for-bandwidth-efficient-gradient-aggregation-in-distributed-cnn-training.pdf)
In this paper, we empirically demonstrate the strong linear correlations between CNN gradients, and propose a gradient vector quantization technique, named GradiVeQ, to exploit these correlations through principal component analysis (PCA) for substantial gradient dimension reduction.

[Gradient Sparsification for Communication-Efficient Distributed Optimization](http://papers.nips.cc/paper/7405-gradient-sparsification-for-communication-efficient-distributed-optimization.pdf)
we propose a convex optimization formulation to minimize the coding length of stochastic gradients. 

[cpSGD: Communication-efficient and differentially-private distributed SGD](http://papers.nips.cc/paper/7984-cpsgd-communication-efficient-and-differentially-private-distributed-sgd.pdf)
distributed stochastic gradient descent is an important subroutine in distributed learning. A setting of particular interest is when the clients are mobile devices, where two important concerns are communication efficiency and the privacy of the clients. we study algorithms that achieve both communication efficiency and differential privacy.

[Snap ML: A Hierarchical Framework for Machine Learning](http://papers.nips.cc/paper/7309-snap-ml-a-hierarchical-framework-for-machine-learning.pdf)
we describe a new software framework for fast training of generalized linear models. The framework, named Snap Machine Learning (Snap ML), combines recent advances in machine learning systems and algorithms in a nested manner to reflect the hierarchical architecture of modern computing systems. 

[Neural Architecture Search with Bayesian Optimisation and Optimal Transport](http://papers.nips.cc/paper/7472-neural-architecture-search-with-bayesian-optimisation-and-optimal-transport.pdf)
conventional BO methods have focused on Euclidean and categorical domains, which, in the context of model selection, only permits tuning scalar hyper-parameters of machine learning algorithms. However, with the surge of interest in deep learning, there is an increasing demand to tune neural network architectures. In this work, we develop NASBOT, a Gaussian process based BO framework for neural architecture search. 

[Sparsified SGD with Memory](http://papers.nips.cc/paper/7697-sparsified-sgd-with-memory.pdf)
In this work we analyze Stochastic Gradient Descent (SGD) with k-sparsification or compression (for instance top-k or random-k) and show that this scheme converges at the same rate as vanilla SGD when equipped with error compensation (keeping track of accumulated errors in memory).

[A theory on the absence of spurious solutions for nonconvex and nonsmooth optimization](http://papers.nips.cc/paper/7511-a-theory-on-the-absence-of-spurious-solutions-for-nonconvex-and-nonsmooth-optimization.pdf)
we study the set of continuous functions that admit no spurious local optima (i.e. local minima that are not global minima) which we term global functions.

[Porcupine Neural Networks: Approximating Neural Network Landscapes](http://papers.nips.cc/paper/7732-porcupine-neural-networks-approximating-neural-network-landscapes.pdf)
In general, the underlying optimization of neural networks is non-convex which makes analyzing their performance challenging. In this paper, we take another approach to this problem by constraining the network such that the corresponding optimization landscape has good theoretical properties without significantly compromising performance. 

[Adding One Neuron Can Eliminate All Bad Local Minima](http://papers.nips.cc/paper/7688-adding-one-neuron-can-eliminate-all-bad-local-minima.pdf)
In this paper, we study the landscape of neural networks for binary classification tasks. Under mild assumptions, we prove that after adding one special neuron with a skip connection to the output, or one special neuron per layer, every local minimum is a global minimum.

[Parameters as interacting particles: long time convergence and asymptotic error scaling of neural networks](http://papers.nips.cc/paper/7945-parameters-as-interacting-particles-long-time-convergence-and-asymptotic-error-scaling-of-neural-networks.pdf)
the performance of neural networks on high-dimensional data distributions suggests that it may be possible to parameterize a representation of a given high-dimensional function with controllably small errors, potentially outperforming standard interpolation methods. We demonstrate, both theoretically and numerically, that this is indeed the case. 

[How SGD Selects the Global Minima in Over-parameterized Learning: A Dynamical Stability Perspective](http://papers.nips.cc/paper/8049-how-sgd-selects-the-global-minima-in-over-parameterized-learning-a-dynamical-stability-perspective.pdf)
the question of which global minima are accessible by a stochastic gradient decent (SGD) algorithm with specific learning rate and batch size is studied from the perspective of dynamical stability

[Distributed Stochastic Optimization via Adaptive SGD](http://papers.nips.cc/paper/7461-distributed-stochastic-optimization-via-adaptive-sgd.pdf)
In this paper, we propose an efficient distributed stochastic optimization method by combining adaptivity with variance reduction techniques. Our analysis yields a linear speedup in the number of machines, constant memory footprint, and only a logarithmic number of communication rounds. 

[On the Local Minima of the Empirical Risk](http://papers.nips.cc/paper/7738-on-the-local-minima-of-the-empirical-risk.pdf)
we consider a general framework which aims to optimize a smooth nonconvex function F, Our objective is to find the ϵ-approximate local minima of the underlying function F while avoiding the shallow local minima.

[Understanding Batch Normalization](http://papers.nips.cc/paper/7996-understanding-batch-normalization.pdf)
Batch normalization (BN) is a technique to normalize activations in intermediate layers of deep neural networks. In this paper we take a step towards a better understanding of BN, following an empirical approach.

[Norm matters: efficient and accurate normalization schemes in deep networks](http://papers.nips.cc/paper/7485-norm-matters-efficient-and-accurate-normalization-schemes-in-deep-networks.pdf)
we present a novel view on the purpose and function of normalization methods and weight-decay, as tools to decouple weights' norm from the underlying optimized objective. 

[LAG: Lazily Aggregated Gradient for Communication-Efficient Distributed Learning](http://papers.nips.cc/paper/7752-lag-lazily-aggregated-gradient-for-communication-efficient-distributed-learning.pdf)
this paper presents a new class of gradient methods for distributed machine learning that adaptively skip the gradient calculations to learn with reduced communication and computation. 

[Automatic differentiation in ML: Where we are and where we should be going](http://papers.nips.cc/paper/8092-automatic-differentiation-in-ml-where-we-are-and-where-we-should-be-going.pdf)
we review the current state of automatic differentiation (AD) for array programming in machine learning (ML), including the different approaches such as operator overloading (OO) and source transformation (ST) used for AD, graph-based intermediate representations for programs, and source languages. Based on these insights, we introduce a new graph-based intermediate representation (IR) which specifically aims to efficiently support fully-general AD for array programming.
